app/stac_api/views.py

```diff
@@ -31,6 +32,17 @@
 logger = logging.getLogger(__name__)
 
 
+def debug_print_request_data(view):
+    if sys.getsizeof(view.request.data) < 500000:
```

Comment:
The first access to `.data` on the request of the `rest_framework` trigger the body stream to be fully read and parsed. Looking at the rest_framework docu (https://www.django-rest-framework.org/api-guide/requests/#data) and at its code. The parsing takes already care of edge cases and malformed body. So I would suggest to remove this check. However once the logging is done, I would quickly test the behavior with a script that would do a POST with a valid json payload e.g. 
```python
import requests
requests.post(url, data ={'hugeString': 'This is a huge String' * 1000 * 1000 * 1000})
```